src/Collection/AllRatesRecord.php

```diff
@@ -10,37 +10,24 @@ class AllRatesRecord extends CountableProviderRecords
 {
     private $base;
 
-    private $time;
+    public function __construct(
+        \Iterator $providerRecords,
+        string $base,
+        int $count,
+        ProviderResource $resource
+    ) {
+        parent::__construct($providerRecords, $count, $resource);
 
-    private $quote;
-
-    private $rate;
-
-    public function __construct(array $time, string $base, array $quote, array $rate)
-    {
         $this->base = $base;
-        $this->time = $time;
-        $this->quote = $quote;
-        $this->rate = $rate;
     }
 
     public function getBase(): string
     {
         return $this->base;
     }
 
-    public function getTime(): array
-    {
-        return $this->time;
-    }
-
-    public function getQuote(): array
-    {
-        return $this->quote;
-    }
-
-    public function getRate(): array
+    public function toAssociativeArray(): array
     {
-        return $this->rate;
+        return iterator_to_array($this);
```

Comment:
This doesn't seem to add any value, and its implementation encourages using array instead of iterators. If the client want to use arrays it can call `iterator_to_array` itself, but Porter and her providers should not encourage this because it loads the entire data set into memory. In this case, since we're not streaming the response, it's already loaded into memory anyway, but it's still a bad idea to encourage this. Recommend removing this method.